src/schema/posts.ts

```diff
@@ -1643,6 +1662,28 @@ export const resolvers: IResolvers<any, Context> = {
       );
       return getPostById(ctx, info, newPost.id);
     },
+    updateSharePost: async (
+      _,
+      {
+        id,
+        sourceId,
+        commentary,
+      }: { id: string; sourceId: string; commentary: string },
+      ctx,
+      info,
+    ): Promise<GQLPost> => {
+      await ctx.con.getRepository(Post).findOneByOrFail({ id });
+      await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
```

Comment:
I think we need some form of test + check if you are eliged to edit this post.
You can see an example in the editPost basically.

Reply:
good point, I was meaning to ask about this -- this should now take that into account as well